File: VB/DataImportExample/TestDataValueConverter.vb
Imports DevExpress.Drawing
Imports DevExpress.Spreadsheet
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace DataImportExample
#Region "#converter"
    Friend Class TestDataValueConverter
        Implements IDataValueConverter
        Private Function IDataValueConverter_TryConvert(value As Object, index As Integer, ByRef result As CellValue) As Boolean Implements IDataValueConverter.TryConvert
            If TypeOf value Is String Then
                Dim strValue As String = TryCast(value, String)

                Try
                    result = DXImage.FromBase64String(strValue)
                Catch
                    Dim str2int As Integer = 0

                    If Int32.TryParse(strValue, str2int) Then
                        result = str2int
                    Else
                        result = strValue
                    End If
                End Try

                Return True
            End If

            result = DevExpress.Spreadsheet.CellValue.TryCreateFromObject(value)
            Return True
        End Function

    End Class
#End Region ' #converter
End Namespace

File: CS/DataImportExample/TestDataValueConverter.cs
using DevExpress.Drawing;
using System;

namespace DataImportExample
{
    #region #converter
    class TestDataValueConverter : DevExpress.Spreadsheet.IDataValueConverter
    {
        public bool TryConvert(object value, int columnIndex, out DevExpress.Spreadsheet.CellValue result)
        {
            switch (value)
            {
                case string strValue:
                    try
                    {
                        result = DXImage.FromBase64String(strValue);
                    }
                    catch
                    {
                        int str2int = 0;
                        if (Int32.TryParse(strValue, out str2int))
                        {
                            result = str2int;
                        }
                        else
                        {
                            result = strValue;
                        }
                    }
                    return true;
                case int intValue:
                    result = intValue;
                    return true;
                case bool boolValue:
                    result = boolValue;
                    return true;
                default:
                    result = value == null ? null : value.ToString();
                    return true;
            }
        }
    }
    #endregion #converter
}
